packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -396,6 +400,25 @@ export class DataStores implements IDisposable {
                 return context.attachState === AttachState.Attached;
             }).map(async ([contextId, context]) => {
                 const contextSummary = await context.summarize(fullTree, trackState);
+                const summary = contextSummary.summary;
+                assert(summary.type === SummaryType.Tree, "context summary is not a tree");
+                const summaryTree = summary.tree[channelsTreeName];
+                if (summaryTree !== undefined) {
```

Comment:
I don't think this check should be here, again because of `disableIsolatedChannels`.
Your code should look something like this:
```ts
const channelsTree = summary.tree[channelsTreeName]?.tree ?? summary.tree;
```
or
```ts
const channelsTree = this.runtime.disableIsolatedChannels ? summary: summary.tree[channelsTreeName];
```
or some other variant of checking; i.e. we can still do the count, even if there's no `.channels` tree.

Reply:
And again, this is _only_ if you plan to count the number of DDS